src/backend/distributed/metadata/pg_get_object_address.c

```diff
@@ -0,0 +1,503 @@
+/*-------------------------------------------------------------------------
+ *
```

Comment:
I think one more generic concern here is that we might need to update this file per PG release. That's why I wonder whether we should name the file as `pg_get_object_address_12_13_14.c` to indicate the supported PG versions.

Also, in general, can we add some more comments on why we need to copy this? Our comments here is too brief to be recognized by readers